src/lib/core/components/computations/Utils.ts

```diff
@@ -7,11 +7,15 @@ export function createComputation(
   overview: ComputationAppOverview,
   displayName: string,
   configuration: unknown,
-  computations: Computation[] = []
+  computations: Computation[] = [],
+  computationId?: string
 ): Computation {
-  const computationId = createNewId(
-    new Set(computations.map((c) => c.computationId))
-  );
+  if (!computationId) {
+    computationId = createNewId(
+      new Set(computations.map((c) => c.computationId))
+    );
+  }
```

Comment:
This can be abbreviated. See [logical nullish assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_nullish_assignment).

```suggestion
    computationId ??= createNewId(
      new Set(computations.map((c) => c.computationId))
    );
```